### PR TITLE
Assorted updates

### PR DESCRIPTION
**Pull request by apasel422**
The most controversial part of this will be 9766baf, which is what I was going for with my comments in #15.

### CODE REVIEW

**Comment by FlashCat** [comment]
r? @reem

(I've picked a reviewer for you, use r? to override)

File: src/lib.rs [hunk 1]
```diff
@@ -93,8 +92,8 @@ impl<K, V, S> LruCache<K, V, S> where K: Eq + Hash, S: HashState {
     ///
     /// cache.insert(1, "a");
     /// cache.insert(2, "b");
-    /// assert_eq!(cache.get(&1), Some(&"a"));
-    /// assert_eq!(cache.get(&2), Some(&"b"));
+    /// assert_eq!(cache.get_mut(&1), Some(&mut "a"));
+    /// assert_eq!(cache.get_mut(&2), Some(&mut "b"));
```
Comment (Gankra) [inline src/lib.rs:96]: `&mut "a"` 0_o

Comment (apasel422) [inline src/lib.rs:96]: Unfortunately there's no `PartialEq` impl that'll work with just `&` when it's within an `Option`.

**Comment by Gankra** [comment]
It's not clear to me that renaming the method to `_mut` is valuable when there is no non-mut variant.

**Comment by apasel422** [comment]
I was just going for consistency, but I can restore it to `get` if you'd prefer.

**Comment by Gankra** [comment]
via @cmr

```
1:58 PM <Elaine> So to me get :: Some Self Thing -> &T, get_mut :: Some Self Thing -> &mut T
1:59 PM <Elaine> get_mut is allowed to borrow refcells mutably etc
1:59 PM <Elaine> the receiver type is an irrelephant.
1:59 PM <Elaine> that's what I expect when I read code
```

So get_mut seems right

**Pull request merged**